### PR TITLE
Checkin onboarding (EXPOSUREAPP-5580)

### PR DESCRIPTION
**Pull request by chris-cwa**

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/TraceLocationSettings.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package de.rki.coronawarnapp.eventregistration
+
+import de.rki.coronawarnapp.eventregistration.storage.TraceLocationPreferences
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class TraceLocationSettings @Inject constructor(val preferences: TraceLocationPreferences) {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/TraceLocationSettings.kt:8]: what is the difference between `TraceLocationSettings` and `TraceLocationPreferences `? 
Can they be one class `TraceLocationSettings` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/onboarding/CheckInOnboardingFragment.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
+class CheckInOnboardingFragment : Fragment(R.layout.fragment_check_in_onboarding), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+
+    @Inject lateinit var settings: TraceLocationSettings
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/onboarding/CheckInOnboardingFragment.kt:21]: Could you please use that from the ViewModel ?

**Comment by mtwalli** [comment]
1- Dark mode card has dimmed colour 
2- Button looks different from other buttons in the App and text colour should be white as per design
3- Badge Text cooler should be white as per design 

device-2021-03-26-155851
device-2021-03-26-155514

File: Corona-Warn-App/src/main/res/values-de/strings.xml [hunk 3]
```diff
@@ -1940,4 +1940,19 @@
...
+    <string name="event_checkin_onboarding_body_consent3">Wenn Sie einen QR-Code einscannen, um sich einzuchecken, wird auf Ihrem Smartphone ein Eintrag für das Event oder den Ort angelegt.</string>
+    <string name="event_checkin_onboarding_body_consent4">Wenn ein zeitgleich mit Ihnen eingecheckter Nutzer später positiv getestet wird und eine Warnung auslöst, werden Sie automatisch anhand der Event-ID gewarnt – auch wenn Ihr Smartphone keine Zufalls-IDs des positiv getesteten Nutzers empfangen hat.</string>
+    <string name="event_checkin_onboarding_body_consent5">Falls Sie später selbst eine Warnung auslösen, werden automatisch alle anderen Nutzer, die zeitgleich mit Ihnen bei einem unter „Meine Check-ins“ eingetragenen Event oder Ort eingecheckt waren, anhand der Event-ID gewarnt.</string>
+    <string name="event_checkin_onboarding_body_consent6">Unter „Meine Check-ins“ können Sie einen Check-in jederzeit löschen. In diesem Fall werden Sie nur gewarnt, wenn Ihr Smartphone die Zufalls-IDs des positiv getesteten Nutzers empfangen und eine Risiko-Begegnung ermittelt hat.</string>
+    <string name="event_checkin_onboarding_body_confirm">Einverstanden</string>
```
[the remark this answers is not in the document]
Comment (chris-cwa) [reply Corona-Warn-App/src/main/res/values-de/strings.xml]: changed it

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/onboarding/CheckInOnboardingFragment.kt (hunk 2 above)
Comment (chris-cwa) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/onboarding/CheckInOnboardingFragment.kt]: moved it

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/TraceLocationSettings.kt (hunk 1 above)
Comment (chris-cwa) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/TraceLocationSettings.kt:8]: i can do it in one class, if you want. generally i like to keep storage (preferences) from domain logic (settings) but maybe i am splitting hairs here....

Comment (chris-cwa) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/TraceLocationSettings.kt]: changed it

**Comment by mtwalli** [comment]
Thanks for addressing the comments, I found two things beside that text has to be updated Ralf already mentioned.

1- Text is cut from both sides
Screenshot 2021-03-30 at 11 01 23

2- the paragraph has to be aligned with the bullet points above as per design
Screenshot 2021-03-30 at 11 02 27

3- A text body is missing `Durch....`
PR
Screenshot 2021-03-30 at 11 09 17

Design
Screenshot 2021-03-30 at 11 09 26

**Comment by ralfgehrer** [comment]
There seems to be minor flaw in the back navigation:
- fresh install (never visited event checkin)
- go to check-in tab bar
- onboarding is shown
- accept
- list of events is show
- use back navigation 

Expected: home screen is  shown
Actual: onboarding is shown

Note: check contact journal onboarding for reference

**Comment by ralfgehrer** [comment]
You could also link the info screen on the "Meine Check-ins" screen's context (...) menu

**Comment by ralfgehrer** [comment]
Screenshot 2021-03-31 at 14 54 42
The second icon differs on Figma

File: Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/check_in_onboarding_stay">
+
+                <TextView
+                    android:id="@+id/check_in_onboarding_card_title"
+                    style="@style/headline6"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="@string/trace_location_onboarding_card_title"
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml:137]: All strings in this constraint layout need to be moved to the legal strings files.

**Review by ralfgehrer — DISMISSED** [review]

**Comment by chris-cwa** [comment]
> There seems to be minor flaw in the back navigation:
> 
> * fresh install (never visited event checkin)
> * go to check-in tab bar
> * onboarding is shown
> * accept
> * list of events is show
> * use back navigation
> 
> Expected: home screen is shown
> Actual: onboarding is shown
> 
> Note: check contact journal onboarding for reference

fixed it!

**Comment by chris-cwa** [comment]
> Screenshot 2021-03-31 at 14 54 42
> The second icon differs from Figma

fixed it!

File: Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content" >
+
+            <ImageView
+                android:id="@+id/check_in_onboarding_image"
+                android:layout_width="0dp"
+                android:layout_height="190dp"
+                android:src="@drawable/checkin_motivation"
```
[the remark this answers is not in the document]
Comment (chris-cwa) [reply Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml]: done

File: Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml (hunk 4 above)
Comment (chris-cwa) [reply Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml:138]: thanks for the hint

**Comment by harambasicluka** [comment]
Minor UI "bug", the grey circle is missing on the second icon:
image

**Comment by mtwalli** [comment]
LGTM - `X` button is missing when it is opened from the menu similar to Contact Dairy Info screen.

- When opened first time from the tab - > no` X `icon 
- When opened from the menu ->` X ` icon is shown

File: Corona-Warn-App/src/main/res/values-bg/strings.xml [hunk 6]
```diff
@@ -1953,16 +1953,6 @@
...
-    <string name="trace_location_onboarding_body_consent4">"Ако друг потребител, регистриран по същото време и от същото място или събитие, изпрати предупреждение, след като бъде диагностициран с коронавирус, Вие ще получите предупреждението въз основа на идентификационния код на събитието дори ако във Вашия смартфон не е получен случаен идентификатор от заразеното лице."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent5">"Ако от своя страна Вие изпратите предупреждение, въз основа на идентификационния код на място или събитие от раздела “Моите регистрации” ще бъдат предупредени всички останали потребители, които са се регистрирали от него по същото време."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent6">"Може да изтриете регистрация от раздела „Моите регистрации“ по всяко време. В този случай само Вие ще получите предупреждение, ако в смартфона Ви има получени случайни ИД на лице, диагностицирано с коронавирус, и е установено потенциално излагане на риск."</string>
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/values-bg/strings.xml:1965]: please only remove it in `values-de/strings.xml`.

File: Corona-Warn-App/src/main/res/values-en/strings.xml [hunk 7]
```diff
@@ -1953,16 +1953,6 @@
...
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent4">"If another user who checked in at the same time is subsequently diagnosed with coronavirus and triggers a warning, you will be warned automatically based on the event ID – even if your smartphone did not receive any random Ids from the infected user."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent5">"If you trigger a warning yourself later, all other users who were checked in to an event or place listed under “My Check-Ins” at the same time are warned based on the event ID."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent6">"You can delete a check-in under “My Check-Ins” at any time. In this case, you will only be warned if your smartphone received the random IDs of a user diagnosed with coronavirus and identified a potential exposure."</string>
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/values-en/strings.xml:1965]: please only remove it in `values-de/strings.xml`.

File: Corona-Warn-App/src/main/res/values-pl/strings.xml [hunk 8]
```diff
@@ -1953,16 +1953,6 @@
...
-    <string name="trace_location_onboarding_body_consent4">"Jeśli u innego użytkownika, który zameldował się w tym samym czasie, zostanie zdiagnozowany koronawirus i uruchomione zostanie ostrzeżenie, otrzymasz automatyczne ostrzeżenie na podstawie identyfikatora wydarzenia – nawet jeśli zarażony użytkownik nie wyśle na Twój smartfon losowego identyfikatora."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent5">"Jeśli w późniejszym czasie uruchomisz ostrzeżenie, wszyscy inni użytkownicy, którzy w tym samym czasie zameldowali swoją obecność w wydarzeniu lub w miejscu wymienionym w obszarze „Moje zameldowania”, otrzymają ostrzeżenie na podstawie identyfikatora wydarzenia."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent6">"W każdej chwili możesz usunąć zameldowanie w obszarze „Moje zameldowania”. W takim przypadku otrzymasz ostrzeżenie tylko wtedy, gdy na Twój smartfon wysłany zostanie losowy identyfikator użytkownika, u którego zdiagnozowano koronawirusa, i zidentyfikowane zostanie potencjalne narażenie."</string>
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/values-pl/strings.xml:1965]: please only remove it in `values-de/strings.xml`.

File: Corona-Warn-App/src/main/res/values-ro/strings.xml [hunk 9]
```diff
@@ -1953,16 +1953,6 @@
...
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent4">"Dacă un alt utilizator care a făcut check-in în același timp este diagnosticat ulterior cu coronavirus și declanșează un avertisment, veți fi avertizat automat pe baza ID-ului de eveniment, chiar dacă smartphone-ul dvs. nu a primit niciun ID aleatoriu de la utilizatorul infectat."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent5">"Dacă ulterior declanșați dvs. un avertisment, toți ceilalți utilizatori care au făcut check-in la evenimentul sau în locul enumerat la „Check-inurile mele” în același timp cu dvs. sunt avertizați pe baza ID-ului de eveniment."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent6">"Puteți oricând șterge un check-in din „Check-inurile mele”. În acest caz, veți fi avertizat doar dacă smartphone-ul dvs. a primit ID-urile aleatorii ale unui utilizator diagnosticat cu coronavirus și a identificat o posibilă expunere."</string>
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/values-ro/strings.xml:1965]: please only remove it in `values-de/strings.xml`.

File: Corona-Warn-App/src/main/res/values-tr/strings.xml [hunk 10]
```diff
@@ -1953,16 +1953,6 @@
...
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent4">"Aynı anda check in yapmış başka bir kullanıcı daha sonra koronavirüs tanısı alırsa ve bir uyarı tetiklerse etkinlik kimliğine göre otomatik olarak uyarı alırsınız; bunun içi akıllı telefonunuzun enfekte olan kullanıcıdan rastgele kimlik almış olması şartı aranmaz."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent5">"Siz daha sonra bir uyarı tetiklerseniz “Check-In’lerim” bölümünde listelenen bir etkinliğe veya yere check in yapmış olan diğer tüm kullanıcılara etkinlik kimliğine göre uyarı gönderilir."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent6">"Bir check-in’i dilediğiniz zaman “Check-In’lerim” bölümünden silebilirsiniz. Silmeniz halinde yalnızca, koronavirüs tanısı almış ve potansiyel maruz kalma durumu belirlenmiş bir kullanıcıdan akıllı telefonunuza rastgele kimlikler gönderilmişse uyarı alırsınız."</string>
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/values-tr/strings.xml:1965]: please only remove it in `values-de/strings.xml`.

File: Corona-Warn-App/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -1967,16 +1967,6 @@
...
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent4">"If another user who checked in at the same time is subsequently diagnosed with coronavirus and triggers a warning, you will be warned automatically based on the event ID – even if your smartphone did not receive any random Ids from the infected user."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent5">"If you trigger a warning yourself later, all other users who were checked in to an event or place listed under “My Check-Ins” at the same time are warned based on the event ID."</string>
-    <!-- YMSG: Onboarding trace location bullet points -->
-    <string name="trace_location_onboarding_body_consent6">"You can delete a check-in under “My Check-Ins” at any time. In this case, you will only be warned if your smartphone received the random IDs of a user diagnosed with coronavirus and identified a potential exposure."</string>
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/values/strings.xml:1979]: please only remove it in `values-de/strings.xml`.

**Review by harambasicluka — COMMENTED** [review]
Please redo the language changes in all files non legal non german files (the ones I highlighted :) )

File: Corona-Warn-App/src/main/res/values-tr/strings.xml (hunk 10 above)
Comment (chris-cwa) [reply Corona-Warn-App/src/main/res/values-tr/strings.xml:1965]: i moved them, if i wouldn't remove them here they would be duplicate

Comment (harambasicluka) [reply Corona-Warn-App/src/main/res/values-tr/strings.xml:1965]: uhh true!!

**Comment by chris-cwa** [comment]
> LGTM - `X` button is missing when it is opened from the menu similar to Contact Dairy Info screen.
> 
> * When opened first time from the tab - > no`X`icon
> * When opened from the menu ->`X` icon is shown

fixed it!

**Review by mtwalli — APPROVED** [review]
LGTM

**Review by LukasLechnerDev — APPROVED** [review]
LGTM

**Review by axelherbstreith — APPROVED** [review]

**Pull request merged**